src/components/Combobox/components/ComboboxItems/ComboboxItems.jsx

```diff
@@ -0,0 +1,124 @@
+import cx from "classnames";
+import { getOptionsByCategories } from "components/Combobox/ComboboxService";
+import {
+  COMBOBOX_CATEGORY_ITEM,
+  COMBOBOX_DIVIDER_ITEM,
+  COMBOBOX_OPTION_ITEM
+} from "components/Combobox/components/ComboboxConstants";
+import { comboboxItemRenderer } from "components/Combobox/components/ComboboxRenderers";
+import { VirtualizedList } from "components";
+import styles from "./ComboboxItems.modules.scss";
+
+export const ComboboxItems = ({
+  categories,
+  options,
+  filterValue,
+  withCategoriesDivider,
+  optionRenderer,
+  activeItemIndex,
+  isActiveByKeyboard,
+  onOptionClick,
+  onOptionEnter,
+  onOptionLeave,
+  optionLineHeight,
+  shouldScrollToSelectedItem,
+  renderOnlyVisibleOptions
+}) => {
+  let items = useMemo(() => {
+    let items = [];
+    if (categories) {
+      const optionsByCategories = getOptionsByCategories(options, categories, filterValue);
+      let optionIndex = 0;
+      Object.keys(optionsByCategories).forEach((categoryId, categoryIndex) => {
+        const withDivider = withCategoriesDivider && categoryIndex !== 0;
+        if (withDivider) items.push({ type: COMBOBOX_DIVIDER_ITEM, height: 17, id: `${categoryId}-divider` });
+        items.push({
+          height: 28,
+          type: COMBOBOX_CATEGORY_ITEM,
+          category: categories[categoryId],
+          id: categoryId,
+          withDivider
+        });
+        const isActive = activeItemIndex === optionIndex;
+        const options = optionsByCategories[categoryId].map(option => {
+          const renderedOption = {
+            type: COMBOBOX_OPTION_ITEM,
+            height: optionLineHeight,
+            belongToCategory: true,
+            index: { optionIndex },
+            option,
+            id: option.id || optionIndex,
+            optionRenderer,
+            isActive,
+            isActiveByKeyboard,
+            onOptionClick,
+            onOptionEnter,
+            onOptionLeave,
+            optionLineHeight,
+            shouldScrollToSelectedItem
+          };
+          optionIndex++;
+          return renderedOption;
+        });
+        items.push(...options);
+      });
+    } else {
+      items = options.map((option, index) => {
+        return {
+          type: COMBOBOX_OPTION_ITEM,
+          index,
+          height: optionLineHeight,
+          option,
+          optionRenderer,
+          id: option.id || index,
+          isActive: activeItemIndex === index,
+          isActiveByKeyboard,
+          onOptionClick,
+          onOptionEnter,
+          onOptionLeave,
+          optionLineHeight,
+          shouldScrollToSelectedItem
+        };
+      });
+    }
+
+    return items;
+  }, [
+    categories,
+    options,
+    filterValue,
+    withCategoriesDivider,
+    activeItemIndex,
+    optionRenderer,
+    isActiveByKeyboard,
+    onOptionClick,
+    onOptionEnter,
+    onOptionLeave,
+    optionLineHeight,
+    shouldScrollToSelectedItem
```

Comment:
lets remove some of these dependencies - so it won't trigger re-calculate of all items on every small change
We can send some of these dependencies in the acutal render function. Lets do it together.